src/main/scala/mesosphere/marathon/core/launchqueue/impl/AppTaskLauncherActor.scala

```diff
@@ -334,55 +349,65 @@ private class AppTaskLauncherActor(
       sender ! MatchedTaskOps(offer.getId, Seq.empty)
 
     case ActorOfferMatcher.MatchOffer(deadline, offer) =>
-      val newTaskOpt: Option[CreatedTask] = taskFactory.newTask(app, offer, tasksMap.values)
-      newTaskOpt match {
-        case Some(CreatedTask(mesosTask, marathonTask)) =>
-          def updateActorState(): Unit = {
-            val taskId = marathonTask.taskId
-            assume(
-              taskId.mesosTaskId == mesosTask.getTaskId,
-              "marathon task id and mesos task id must be equal"
-            )
-
-            tasksMap += taskId -> marathonTask
-            inFlightTaskLaunches += taskId -> None
-            tasksToLaunch -= 1
-            OfferMatcherRegistration.manageOfferMatcherStatus()
-          }
+      val taskOp: Option[TaskOp] = taskOpLogic.inferTaskOp(app, offer, tasksMap.values)
+      taskOp match {
+        case Some(launch: TaskOp.Launch) => handleLaunchOp(launch, offer)
+        case Some(reserve: TaskOp.ReserveAndCreateVolumes) => handleReserveOp(reserve, offer)
+        case None => sender() ! MatchedTaskOps(offer.getId, Seq.empty)
+      }
+  }
 
-          updateActorState()
+  private[this] def handleLaunchOp(launch: TaskOp.Launch, offer: Mesos.Offer): Unit = {
+    log.info("Request: Launch task '{}', version '{}'. {}", launch.taskId.idString, app.version, status)
 
-          log.info("Request to launch task with id '{}', version '{}'. {}",
-            mesosTask.getTaskId.getValue, app.version, status)
+    updateActorState(launch.newTask, Some(launch.taskInfo))
+    self ! AppTaskLauncherActor.ScheduleTaskOpNotificationTimeout(launch)
+    sender() ! MatchedTaskOps(offer.getId, Seq(TaskOpWithSource(myselfAsLaunchSource, launch)))
+  }
 
-          val launch = OfferMatcher.Launch(mesosTask, marathonTask)
-          self ! AppTaskLauncherActor.ScheduleTaskLaunchNotificationTimeout(launch)
-          sender() ! MatchedTaskOps(offer.getId, Seq(TaskOpWithSource(myselfAsLaunchSource, launch)))
+  private[this] def handleReserveOp(reserve: TaskOp.ReserveAndCreateVolumes, offer: Mesos.Offer): Unit = {
```

Comment:
Just noting that handleLaunch and handleReserveOp do exactly the same (other than a different log message). Do you think we can be more DRY here?